server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -8,28 +8,349 @@
+            "fake desc",
+            "name",
+            "secondExtension",
+            "version",
+            "1.0",
+            "opensearch.version",
+            Version.CURRENT.toString(),
+            "java.version",
+            System.getProperty("java.specification.version"),
+            "classname",
+            "FakeExtensions"
+        );
+
+        Settings settings = Settings.builder().build();
+        ExtensionsOrchestrator orchestrator = new ExtensionsOrchestrator(settings, extensionsDir);
 
-        ExtensionsSettings extensions = ExtensionsOrchestrator.readFromExtensionsYml("/config/extensions.yml");
+        Set<DiscoveryExtension> expectedExtensionsSet = new HashSet<DiscoveryExtension>();
 
-        assertNotNull(extensions);
-        assertEquals("firstExtension", extensions.getExtensions().get(0).getName());
-        assertEquals("uniqueid1", extensions.getExtensions().get(0).getEphemeralId());
-        assertEquals("myIndependentPluginHost1", extensions.getExtensions().get(0).getHostName());
-        assertEquals("127.0.0.0", extensions.getExtensions().get(0).getHostAddress());
-        assertEquals("9301", extensions.getExtensions().get(0).getPort());
-        assertEquals("2.0.0", extensions.getExtensions().get(0).getVersion());
-        assertEquals("secondExtension", extensions.getExtensions().get(1).getName());
-        assertEquals("uniqueid2", extensions.getExtensions().get(1).getEphemeralId());
-        assertEquals("myIndependentPluginHost2", extensions.getExtensions().get(1).getHostName());
-        assertEquals("127.0.0.0", extensions.getExtensions().get(1).getHostAddress());
-        assertEquals("9301", extensions.getExtensions().get(1).getPort());
-        assertEquals("2.0.0", extensions.getExtensions().get(1).getVersion());
+        expectedExtensionsSet.add(
+            new DiscoveryExtension(
+                "firstExtension",
+                "id",
+                "uniqueid1",
+                "myIndependentPluginHost1",
+                "127.0.0.0",
+                new TransportAddress(TransportAddress.META_ADDRESS, 9300),
+                new HashMap<String, String>(),
+                Version.fromString("3.0.0"),
+                PluginInfo.readFromProperties(pluginDir1)
+            )
+        );
 
+        expectedExtensionsSet.add(
+            new DiscoveryExtension(
+                "secondExtension",
+                "id",
+                "uniqueid2",
+                "myIndependentPluginHost2",
+                "127.0.0.1",
+                new TransportAddress(TransportAddress.META_ADDRESS, 9301),
+                new HashMap<String, String>(),
+                Version.fromString("2.0.0"),
+                PluginInfo.readFromProperties(pluginDir2)
+            )
+        );
+
+        assertEquals(expectedExtensionsSet, orchestrator.extensionsSet);
+    }
+
+    public void testNonAccessibleDirectory() throws Exception {
+        Settings settings = Settings.builder().build();
```

Comment:
Can we create dummy settings here instead of empty?

Reply:
What would you recommend as a replacement?

Reply:
Something like `Settings.builder().put("node.name", "my-node").build();`